### PR TITLE
Refactoring slices in Medication & Immunization profiles to slice by value set (FHIR-46391)

### PR DESCRIPTION
**Pull request by dbojicic**
Related to AU Base changes in https://github.com/hl7au/au-fhir-base/pull/848. Fixes https://jira.hl7.org/browse/FHIR-46391

Changed: 
- AU Core Immunization
- AU Core Medication
- AU Core MedicationRequest

### CODE REVIEW

File: input/resources/au-core-medication.xml [hunk 1]
```diff
@@ -69,6 +69,11 @@
       <path value="Medication.code.coding"/>
       <sliceName value="pbs"/>
       <mustSupport value="true"/>
+      <max value="1" />
```
Comment (dt-r) [inline input/resources/au-core-medication.xml:72]: Max  value of 1 is not appropriate for the terminology slices in AU Core Medication and AU Core MedicationRequest - AU Base does not restrict to be max of 1.

File: input/resources/au-core-medicationrequest.xml [hunk 2]
```diff
@@ -121,6 +121,11 @@
       </extension>
       <path value="MedicationRequest.medication[x].coding"/>
       <sliceName value="pbs"/>
+      <max value="1" />
```
Comment (dt-r) [inline input/resources/au-core-medicationrequest.xml:124]: Max  value of 1 is not appropriate for the terminology slices in AU Core Medication and AU Core MedicationRequest - AU Base does not restrict to be max of 1.

**Review by dt-r — CHANGES_REQUESTED** [review]
See inline comments. 
Also add to change log - this change may be significant enough that although this is compatible substantive change it may require coding changes in systems.

File: input/resources/au-core-medication.xml (hunk 1 above)
Comment (dbojicic) [reply input/resources/au-core-medication.xml]: Fixed

File: input/resources/au-core-medicationrequest.xml (hunk 2 above)
Comment (dbojicic) [reply input/resources/au-core-medicationrequest.xml]: Fixed

**Comment by dbojicic** [comment]
> See inline comments. Also add to change log - this change may be significant enough that although this is a compatible substantive change it may require coding changes in systems. 

Applied.

File: input/pagecontent/changes.md [hunk 3]
```diff
@@ -41,6 +42,7 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from MedicationRequest.dispenseRequest, MedicationRequest.dispenseRequest.validityPeriod, MedicationRequest.dispenseRequest.numberOfRepeatsAllowed, MedicationRequest.dispenseRequest.quantity [FHIR-45088](https://jira.hl7.org/browse/FHIR-45088)
   - removed Must Support from MedicationRequest.substitution, MedicationRequest.substitution.allowed[x] [FHIR-45088](https://jira.hl7.org/browse/FHIR-45088)
   - added Must Support to MedicationRequest.reasonReference [FHIR-45090](https://jira.hl7.org/browse/FHIR-45090)
+  - changed medicationCodeableConcept.coding slicing rule from slicing by system to slice by value set [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:45]: Missing first part of FHIR Path. changed 'Medicationrequest.medicationCodeableConcept'

File: input/pagecontent/changes.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from Immunization.protocolApplied, Immunization.protocolApplied.series, Immunization.protocolApplied.targetDisease, Immunization.protocolApplied.doseNumber[x] [FHIR-44674](https://jira.hl7.org/browse/FHIR-44674), [FHIR-44656](https://jira.hl7.org/browse/FHIR-44656), [FHIR-44654](https://jira.hl7.org/browse/FHIR-44654)
   - removed Must Support from Immunization.reasonCode [FHIR-44654](https://jira.hl7.org/browse/FHIR-44654), [FHIR-45968](https://jira.hl7.org/browse/FHIR-45968)
   - added Must Support to Immunization.primarySource [FHIR-44659](https://jira.hl7.org/browse/FHIR-44659)
+  - changed Immunization.vaccineCode slicing rule from slicing by system to slice by value set [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:30]: Suggest wording change to:
support for AIR codes and AMT Vaccine codes in Immunization.vaccineCode changed from slicing by value:system to value set. These slices are now defined in AU Core and not inherited from AU Base (see related AU Base change [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821)).

File: input/pagecontent/changes.md [hunk 5]
```diff
@@ -49,6 +51,7 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from Medication.ingredient.isActive [FHIR-45220](https://jira.hl7.org/browse/FHIR-45220), [FHIR-45130](https://jira.hl7.org/browse/FHIR-45130), [FHIR-45049](https://jira.hl7.org/browse/FHIR-45049)
   - removed Must Support from Medication.ingredient.strength [FHIR-45220](https://jira.hl7.org/browse/FHIR-45220)
   - removed Must Support from Medication.amount [FHIR-45219](https://jira.hl7.org/browse/FHIR-45219)
+  - changed medicationCodeableConcept.coding slicing rule from slicing by system to slice by value set [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:54]: element FHIR Path wrong for this profile and similar wording change to above:
support for AMT codes and PBS item codes in Medication.code changed to define slicing in AU Core (no longer inherited from AU Base as per [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821)). Slicing discriminator changed from slicing by value:system to value set and AMT value set binding is required within the AMT slice. [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)

**Review by dt-r — CHANGES_REQUESTED** [review]
See inline comments

File: input/pagecontent/changes.md (hunk 3 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: Fixed

File: input/pagecontent/changes.md (hunk 4 above)
Comment (dbojicic) [reply input/pagecontent/changes.md:30]: Fixed

File: input/pagecontent/changes.md (hunk 5 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: Fixed

File: input/pagecontent/changes.md [hunk 6]
```diff
@@ -27,6 +27,8 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from Immunization.protocolApplied, Immunization.protocolApplied.series, Immunization.protocolApplied.targetDisease, Immunization.protocolApplied.doseNumber[x] [FHIR-44674](https://jira.hl7.org/browse/FHIR-44674), [FHIR-44656](https://jira.hl7.org/browse/FHIR-44656), [FHIR-44654](https://jira.hl7.org/browse/FHIR-44654)
   - removed Must Support from Immunization.reasonCode [FHIR-44654](https://jira.hl7.org/browse/FHIR-44654), [FHIR-45968](https://jira.hl7.org/browse/FHIR-45968)
   - added Must Support to Immunization.primarySource [FHIR-44659](https://jira.hl7.org/browse/FHIR-44659)
+  - changed Immunization.vaccineCode slicing rule from slicing by system to slice by value set [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
+  - support for AIR codes and AMT Vaccine codes in Immunization.vaccineCode changed to define the slices in AU Core (no longer inherited from AU Base as per [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821)). Slicing discriminator changed from slicing by value:system to slicing by value set and AMT value set binding is required within the AMT Vaccine slice . [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:31]: 1. Incorrect spacing here, this has ended up split over two bullet points. 
2. Last sentence has extra whitespace before the full stop.

File: input/pagecontent/changes.md [hunk 7]
```diff
@@ -42,7 +43,7 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from MedicationRequest.dispenseRequest, MedicationRequest.dispenseRequest.validityPeriod, MedicationRequest.dispenseRequest.numberOfRepeatsAllowed, MedicationRequest.dispenseRequest.quantity [FHIR-45088](https://jira.hl7.org/browse/FHIR-45088)
   - removed Must Support from MedicationRequest.substitution, MedicationRequest.substitution.allowed[x] [FHIR-45088](https://jira.hl7.org/browse/FHIR-45088)
   - added Must Support to MedicationRequest.reasonReference [FHIR-45090](https://jira.hl7.org/browse/FHIR-45090)
-  - changed medicationCodeableConcept.coding slicing rule from slicing by system to slice by value set [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
+  - support for Australian Medication (AMT) codes and PBS Item Codes in MedicationRequest.medicationCodeableConcept changed to define slicing in AU Core (no longer inherited from AU Base as per [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821)). Slicing discriminator changed from slicing by value:system to slicing by value set and value set binding is required within the AMT and PBS slices. [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:46]: Australian Medication should read  Australian Medicines Terminology but recommend to just say "AMT" in place of "Australian Medication (AMT)" - this comment applies to two locations in this file.

**Review by dt-r — CHANGES_REQUESTED** [review]
see inline comments

File: input/pagecontent/changes.md (hunk 6 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: 1. No changes made; seems to display fine now - it's on a single bullet point, split over two lines
2. Fixed 
3. Fixed

File: input/pagecontent/changes.md (hunk 7 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: Fixed

File: input/pagecontent/changes.md [hunk 8]
```diff
@@ -49,6 +52,7 @@ This change log documents the significant updates and resolutions implemented fr
   - removed Must Support from Medication.ingredient.isActive [FHIR-45220](https://jira.hl7.org/browse/FHIR-45220), [FHIR-45130](https://jira.hl7.org/browse/FHIR-45130), [FHIR-45049](https://jira.hl7.org/browse/FHIR-45049)
   - removed Must Support from Medication.ingredient.strength [FHIR-45220](https://jira.hl7.org/browse/FHIR-45220)
   - removed Must Support from Medication.amount [FHIR-45219](https://jira.hl7.org/browse/FHIR-45219)
+  - support for AMT codes and PBS Item Codes in Medication.medicationCodeableConcept changed to define slicing in AU Core (no longer inherited from AU Base as per [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821)). Slicing discriminator changed from slicing by value:system to value set and the binding strength has been corrected to required. [FHIR-46391](https://jira.hl7.org/browse/FHIR-46391)
```
Comment (dt-r) [inline input/pagecontent/changes.md:55]: so close, element name is not 'medicationCodeableConcept' in the medication resource, this is just Medication.code.

**Review by dt-r — CHANGES_REQUESTED** [review]
last one

File: input/pagecontent/changes.md (hunk 8 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: fixed

**Review by dt-r — APPROVED** [review]

**Pull request merged**